universum/modules/structure_handler.py

```diff
@@ -230,6 +252,12 @@ def execute_step_structure(self, configs: Configuration, step_executor) -> None:
         if self.active_background_steps:
             self.run_in_block(self.report_background_steps, "Reporting background steps", False)
 
+    def _build_next_step_name(self, name):
```

Comment:
The name of the function suggests that it doesn't have side effects.
Either the name or the logic has to be fixed.